ocrd_anybaseocr/cli/ocrd_anybaseocr_block_segmentation.py

```diff
@@ -112,53 +124,137 @@ def process(self):
         results = mrcnn_model.detect([img_array], verbose=1)    
         r = results[0]        
         
-        page_xywh['features'] += ',blksegmented'
         
+        # define reading order on basis of coordinates
+        reading_order = []
         for i in range(len(r['rois'])):                
             
+            width,height,_ = img_array.shape
+            min_x = r['rois'][i][0]
+            min_y = r['rois'][i][1]
+            reading_order.append((min_y, min_x))
+            
+        reading_order = sorted(reading_order, key=lambda reading_order:(reading_order[0], reading_order[1]))
+        
+        
+        #Creating Reading Order object in PageXML
+        order_group = OrderedGroupType(caption="Regions reading order",id=page_id)
+        reading_order_object = ReadingOrderType()
+        for i in range(len(r['rois'])):
+            min_x = r['rois'][i][0]
+            min_y = r['rois'][i][1]
+            max_x = r['rois'][i][2]
+            max_y = r['rois'][i][3]
+            region_polygon = [[min_x, min_y], [max_x, min_y], [max_x, max_y], [min_x, max_y]]
+            
+            cut_region_polygon = border.intersection(Polygon(region_polygon))
+            if cut_region_polygon.is_empty:
+                continue
+                
+            order_index = reading_order.index((min_y, min_x))
+            region_id = '%s_region%04d' % (page_id, order_index)
+            regionRefIndex = RegionRefIndexedType(index=order_index, regionRef=region_id)
+            order_group.add_RegionRefIndexed(regionRefIndex)
+            
+        reading_order_object = ReadingOrderType()
+        reading_order_object.set_OrderedGroup(order_group)
+        page.set_ReadingOrder(reading_order_object)
+        
+        
+        for i in range(len(r['rois'])):                
             width,height,_ = img_array.shape
             min_x = r['rois'][i][0]
             min_y = r['rois'][i][1]
             max_x = r['rois'][i][2]
             max_y = r['rois'][i][3]
+            region_polygon = [[min_x, min_y], [max_x, min_y], [max_x, max_y], [min_x, max_y]]
+            
+            cut_region_polygon = border.intersection(Polygon(region_polygon))
+            if cut_region_polygon.is_empty:
+                continue
+            cut_region_polygon = [i for i in zip(list(cut_region_polygon.exterior.coords.xy[0]),list(cut_region_polygon.exterior.coords.xy[1]))][:-1]
+            #print(cut_region_polygon)
+            region_polygon = coordinates_for_segment(cut_region_polygon, page_image, page_xywh)
+            region_points = points_from_polygon(region_polygon)
+            
+            read_order = reading_order.index((min_y, min_x))
             
             #small post-processing incase of paragrapgh to not cut last alphabets
             if (min_x - 5) > width and r['class_ids'][i] == 2:
                 min_x-=5
             if (max_x + 10) < width and r['class_ids'][i] == 2:
                 min_x+=10
```

Comment:
That's too late here! You already converted the bbox to `region_points` above. So you derived image and its coordinates will be inconsistent. Move that section upwards before L197.